networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/LibP2PPeer.java

```diff
@@ -86,6 +86,9 @@ public LibP2PPeer(
 
   @Override
   public void checkPeerIdentity() {
+    if (maybeAgentString.isEmpty()) {
+      return;
+    }
```

Comment:
I think that's backwards and now we'll never request the peer identity.

Reply:
Opps